### PR TITLE
Disable warning C4251 and C4275 in MSVC

### PR DESCRIPTION
**Pull request by duxiuxing**
I create a Win32 DLL project in Visual Studio 2010 with gtest and gmock:
image

A lots of C4251 and C4275 warnings came out after the project was compiled:
image

I am a professional programmer, I always treat warnings as errors. So I use the GTEST_DISABLE_MSC_WARNINGS_PUSH_() and GTEST_DISABLE_MSC_WARNINGS_POP_() macros to disable all warning C4251 and warning C4275.

Now the whole world is clean.

### CODE REVIEW

**Comment by duxiuxing** [comment]
Hello Google!!!

**Comment by gennadiycivil** [comment]
@duxiuxing Thank you for your contribution. Could you please provide explanation why this change is needed. Visual Studio 2010 is very old, we would not accept changes solely focusing on this compiler/IDE.
How would this change benefit the entire googletest community?

**Comment by KindDragon** [comment]
VS2015/2017 has same warnings PR 862

File: googlemock/include/gmock/gmock-cardinalities.h [hunk 1]
```diff
@@ -120,8 +120,16 @@ class GTEST_API_ Cardinality {
   static void DescribeActualCallCountTo(int actual_call_count,
                                         ::std::ostream* os);
 
+#if GTEST_NEED_DLL_DECL
```
Comment (KindDragon) [inline googlemock/include/gmock/gmock-cardinalities.h]: I think better create separate macros that will be decalred based on your macros `GTEST_NEED_DLL_DECL`

Comment (gennadiycivil) [reply googlemock/include/gmock/gmock-cardinalities.h]: I am going to close this PR. If this problem is still important enough @KindDragon and @duxiuxing could collaborate to create a better solution, perhaps based on oldr #862

**Pull request closed without merging**